### PR TITLE
(maint) Remove Configurer::DownloaderFactory

### PR DESCRIPTION
**Pull request by Magisus**
This commit removes the DownloaderFactory, which was an example of
unnecessary factory pattern paired with unused dependency injection.
This updates the Configurer constructor to no longer accept a factory as
an argument, which should be a private API change. If this turns out to
not be the case, this commit can be reverted.

### CODE REVIEW

File: lib/puppet/configurer.rb [hunk 1]
```diff
@@ -56,7 +55,7 @@ def init_storage
     end
   end
 
-  def initialize(factory = Puppet::Configurer::DownloaderFactory.new, transaction_uuid = nil, job_id = nil)
+  def initialize(transaction_uuid = nil, job_id = nil)
```
Comment (melissa) [inline lib/puppet/configurer.rb]: we aren't worried about this interface changing? It's not going to break things for users?

File: lib/puppet/configurer/plugin_handler.rb [hunk 2]
```diff
@@ -4,15 +4,33 @@
 require 'puppet/configurer'
 
 class Puppet::Configurer::PluginHandler
-  def initialize(factory)
```
Comment (melissa) [inline lib/puppet/configurer/plugin_handler.rb]: Shouldn't all classes have an init, even if it doesn't really do anything?

File: lib/puppet/configurer.rb (hunk 1 above)
Comment (MikaelSmith) [reply lib/puppet/configurer.rb]: I have minor concerns about it. Traditionally we'd do this in a Puppet 6. We could create a Puppet 6 branch now and retarget it there.

Comment (Magisus) [reply lib/puppet/configurer.rb]: @branan was similarly minorly concerned, but also said he considers this private API. His recommendation was to merge in 5.4.0 and revert if it causes issues somewhere, but I would also be fine saving this for Puppet 6 (though in that case I'd like to branch and still merge, so this doesn't get lost).

**Pull request merged**